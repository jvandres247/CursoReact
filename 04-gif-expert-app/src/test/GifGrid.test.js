import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GifGrid } from "../components/GifGrid";
import { useFetchGifs } from "../hooks/useFetchGifs";
jest.mock("../hooks/useFetchGifs");

describe("Pruebas en <GifGrid/>", () => {
  const category = "naruto";

  test("Debe de mostrarse correctamente", () => {
    useFetchGifs.mockReturnValue({
        data: [],
        loading: true
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar items cuando se cargan imagenes useFetchGifs", () => {
    const gifs = [{
        id: 'ABC',
        url: 'https://www.google.com',
        title: 'Algo'
    },{
        id: 'DEF',
        url: 'https://www.google.com',
        title: 'Algo Dos'
    }]
    useFetchGifs.mockReturnValue({
        data: gifs,
        loading: false
    });
    const wrapper = shallow(<GifGrid category={category} />);
    //expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
