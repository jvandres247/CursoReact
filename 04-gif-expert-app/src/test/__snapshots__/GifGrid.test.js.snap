// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid/> Debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3>
    naruto
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Algo"
      url="https://www.google.com"
    />
    <GifGridItem
      id="DEF"
      key="DEF"
      title="Algo Dos"
      url="https://www.google.com"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid/> Debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3>
    naruto
  </h3>
  <p
    className="card animate__animated animate__flash"
  >
    Loading....
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
